### PR TITLE
9 taser and ammo switching

### LINKED ISSUE #9: Taser and ammo switching
# Problem statement

The game consists of not only masks (that are not really ammo but is treated as it), but also the taser ammo, which will be activated using the right click of the mouse. The taser is limited in the sense that it has an energy level that will decrease each time it's used. The energy will recharge over time but will only shoot if it has the necessary energy amount. 

Implement the taser with the energy level and include the behaviour into the clickable interface defined in #1.

# Definition of done

- We can not only shoot masks but also tase people, freezing them

### PR DESCRIPTION
**Pull request by JuanJoseJ**
This code is a controller for an in-game tool with a limited charge. The taser can be used and the charge decreases with each use. The charge increases slightly every second. The controller has variables for the taser battery, cost, maximum battery, and charge per second. The taser can be activated by right-clicking on it if the trigger is free and there is enough charge. The battery charge is displayed in the console. The taser recharges itself over time and the recharge rate is also displayed.

Closes #9

### CODE REVIEW

**Review by iSergioMejia — COMMENTED** [review]
Looks good to me. However, I don't know if this also has to be handled like #10 as a singleton. @theenergici what do you think?

File: GameDesign/Assets/Scripts/TaserController.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class TaserController : MonoBehaviour{
```
Comment (iSergioMejia) [inline GameDesign/Assets/Scripts/TaserController.cs:5]: Can we change it to `TaserManager` to be aligned with other classes?

**Review by iSergioMejia — CHANGES_REQUESTED** [review]
We need to integrate this behaviour to the new ClickManager that is being created in #10 and leave this only as the state manager for the UI to contact. For now it's good like this to merge as soon as the requested changes are done

**Comment by theenergici** [comment]
> Looks good to me. However, I don't know if this also has to be handled like #10 as a singleton. @theenergici what do you think?

tbh, in theory. maybe we should have it as a singleton to make sure its not called multiple times in the scene. but I don't think is necessary.

**Review by theenergici — APPROVED** [review]
should work, in the future if we decide to change the input system, it will  have to be adjusted.
the idea of making it a singleton to ensure that there is only one in the scene may prove good. but is not necessary.
it think this is fine.

**Review by Sebasgut1 — APPROVED** [review]

File: GameDesign/Assets/Scripts/TaserController.cs (hunk 1 above)
Comment (JuanJoseJ) [reply GameDesign/Assets/Scripts/TaserController.cs:5]: **_### No_**

**Review by iSergioMejia — APPROVED** [review]

**Pull request merged**